tests/suites/test_suite_pk.function

```diff
@@ -964,6 +965,51 @@ void valid_parameters()
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_PK_PARSE_C:PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC */
+void test_bitlen_setup(data_t *priv_key, data_t *pub_key, int pk_type, int curve_or_keybits)
+{
+    mbedtls_pk_context pk, pk_pub, new_pk;
+    size_t bitlen;
+
+    mbedtls_svc_key_id_t pub_key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    if ((pk_type == MBEDTLS_PK_RSA) || (pk_type == MBEDTLS_PK_RSASSA_PSS)) {
+        bitlen = curve_or_keybits;
+    } else if (pk_type == MBEDTLS_PK_ECDSA) {
+        TEST_ASSERT(mbedtls_ecc_group_to_psa(curve_or_keybits, &bitlen) != 0);
+    } else {
+        TEST_FAIL("Key type not supported");
+    }
+
+    PSA_INIT();
+
+    mbedtls_pk_init(&pk);
+    mbedtls_pk_init(&pk_pub);
+    mbedtls_pk_init(&new_pk);
+
+    TEST_EQUAL(pk_setup(&pk, pk_type, (size_t) curve_or_keybits), 0);
+
+    TEST_ASSERT(mbedtls_pk_parse_key(&pk, priv_key->x, priv_key->len, NULL, 0) == 0);
```

Comment:
Wait, unless I'm missing something here you are initializing `pk` twice in a row (`pk_setup` + `mbedtls_pk_parse_key`). I'm a bit surprised that this actually works since the 2nd call should definitely fail because `pk` is already initialized. @mpg what am I missing here?